server/routes/courtRegister/courtRegisterController.ts

```diff
@@ -36,13 +38,25 @@ export default class CourtRegisterController {
 
   async showAllCourtsPaged(req: Request, res: Response): Promise<void> {
     const page = parseInt(req.query.page as string, 10) || 1
-    const courtsPage = await this.courtRegisterService.getPageOfCourts(context(res), page - 1, 40)
+    const filter = this.parseFilter(req)
+    const courtsPage = await this.courtRegisterService.getPageOfCourts(context(res), page - 1, 40, filter)
 
-    const view = new AllCourtsPagedView(courtsPage)
+    const view = new AllCourtsPagedView(courtsPage, filter)
 
     res.render('pages/court-register/allCourtsPaged', view.renderArgs)
   }
 
+  parseFilter(req: Request): AllCourtsFilter {
+    const activeString: string = req.query.active as string
+    let active = null
+    if (activeString === 'true') active = true
+    else if (activeString === 'false') active = false
+    let courtTypeIds: string[] = req.query.courtTypeIds as string[]
+    if (courtTypeIds === undefined) courtTypeIds = null
+    else if (!Array.isArray(courtTypeIds)) courtTypeIds = [courtTypeIds]
```

Comment:
not sure I'm understanding this.  You're casting to an array on line 54 and then checking to see if it is an array here - won't this only be if it is `null`?

Reply:
If there is a single value then the query parameter is a `string` rather than a `string[]` hence the above.  Feels wrong to me too, I'll look for a better way.